src/medals/utils.cljs

```diff
@@ -2,12 +2,29 @@
   (:require [reagent.core :as r]
             [medals.constants :refer [ALLOWED_SUMMATION_VALUES]]))
 
-
 (defn get-total-count [country]
-  (def filtered-count (r/atom []))
-  (for [item (keys country)]
-    (when (contains? ALLOWED_SUMMATION_VALUES item)
-      (println  filtered-count (get country key)))))
+  (let [filtered-count (r/atom [])]
+    (doseq [item (keys country)]
+      (when (contains? ALLOWED_SUMMATION_VALUES item)
+        (reset! filtered-count (conj @filtered-count (get country item)))))
+    (reduce + @filtered-count)))
+
+(defn sort-countries [countries,sort-by]
+  (sort (fn [countryA countryB] (cond
+                                  (> (get countryA sort-by) (get countryB sort-by)) -1
+                                  (< (get countryA sort-by) (get countryB sort-by)) 1
+                                  :else (case sort-by
+                                          :total (if (> (:gold countryA) (:gold countryB)) -1 1)
+                                          :gold (if (>  (:silver countryA) (:silver countryB)) -1 1)
+                                          :silver (if (> (:gold countryA) (:gold countryB)) -1 1)
+                                          :bronze (if (> (:gold countryA) (:gold countryB)) -1 1)))) countries))
+
```

Comment:
[juxt](https://clojuredocs.org/clojure.core/juxt) is a very powerful function. You can solve this in much fewer lines of code using [sort-by](https://clojuredocs.org/clojure.core/sort-by) and juxt. You can see the example in the sort-by link

Reply:
Acknowledged. I have read about it but I didn't get into the depth of **juxt** but I will now. Thankyou